src/v/kafka/server/offset_translator.h

```diff
@@ -53,10 +53,21 @@ class offset_translator {
 private:
     int64_t delta(model::offset o) const {
         auto it = _cfg_mgr.lower_bound(o);
-        if (it != _cfg_mgr.begin()) {
-            --it;
+        if (it == _cfg_mgr.begin()) {
+            if (it->first == o) {
+                return it->second.idx();
+            }
+            /**
+             * iterator points to the first configuration with offset greater
+             * than then requsted one. Knowing an index of that configuration we
+             * know that there was exactly (index -1) configurations with offset
+             * lower than the current one. We can simply subtract one from
+             * index.
+             */
+            return std::max<int64_t>(0, it->second.idx() - 1);
         }
-        return it->second.idx();
+
+        return std::prev(it)->second.idx();
```

Comment:
When `it != _cfg_mgr.begin()` it might be equal to `_cfg_mgr.begin() + 1` in which case the return value `std::prev(it)->second.idx();` is equivalent to `_cfg_mgr.begin()->second.idx()` which also happens to be the same thing that is returned when `it` was originally `_cfg_mgr.begin()` _and_ `it->first == o`. this ends up being pretty confusing. it might be useful to write a comment for each case being handled and/or eliminate this redundancy.

edit: the comment right above this about `iterator points to the first...` is really helpful and clear. maybe repeat something like that for the other cases in this method?

Reply:
done